### PR TITLE
🤖 Compute rMATs readlength inputs

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

To ease the burden on OPs, we've introduced the following quality of life changes for the RNAseq workflow:
- rmats_read_length is no longer a required input
- rmats_variable_read_length no longer has a default value
- The rMATs workflow now has a tool that will poll the first 1000000 reads from the input bam and report the each read length observed as well as how many times that read length is observed. The tool will then report two values: first, the read length is defined as the read length that was most observed; second, if more than one read length is observed it will return variable read length as true.
- If the user wishes to override either of the values calculated by this tool, they can provide the values at workflow input. The workflow will always defer to user provided values and only use the calculated values when no user input is provided.

Closes https://github.com/d3b-center/bixu-tracker/issues/1864

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/d61ea502-b8a3-442d-83f3-5ee20d1d5a70/#

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

File: tools/samtools_readlength_bam.cwl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+  - class: DockerRequirement
+    dockerPull: 'pgc-images.sbgenomics.com/d3b-bixu/samtools:1.9'
+baseCommand: [samtools,view]
+arguments:
+  - position: 1
+    shellQuote: false
+    valueFrom: >-
+      $(inputs.input_bam.path) | head -n 1000000 | cut -f 10 | perl -ne 'chomp;print length($_) . "\n"' | sort | uniq -c | sort -nr -k1,1 > $(inputs.input_bam.nameroot).bam_readlength
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/samtools_readlength_bam.cwl:16]: No. The first sort would convert
```
151
151
150
151
150
```
into
```
150
150
151
151
151
```
Then uniq -c would convert that to:
```
2 150
3 151
```
Then the second sort would convert that to:
```
3 151
2 150
```

File: tools/samtools_readlength_bam.cwl [hunk 2]
```diff
@@ -0,0 +1,45 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: samtools_readlength_bam
+doc: |-
+  samtools view file.bam | head -n 1000000 | cut -f 10 | perl -ne 'chomp;print length($_) . "\n"' | sort | uniq -c | sort -nr -k1,1
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/samtools_readlength_bam.cwl:5]: Will do

File: tools/seqkit_readlength_fastq.cwl [hunk 3]
```diff
@@ -0,0 +1,47 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: seqkit_readlength_fastq
+doc: |-
+  seqkit fx2tab -nl NA18152.fastq
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/seqkit_readlength_fastq.cwl:5]: Will do.

**Review by sakshamphul — APPROVED** [review]

File: tools/seqkit_readlength_fastq.cwl (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/seqkit_readlength_fastq.cwl]: I get your point. Perhaps we should include sample command lines in our docs moving forward.

**Pull request merged**